src/Cronofy/AddToCalendarRequestBuilder.cs

```diff
@@ -0,0 +1,109 @@
+        private UpsertEventRequestBuilder upsertEventRequestBuilder;
+
+        /// <summary>
+        /// The event details for the request.
+        /// </summary>
+        private UpsertEventRequest upsertEventRequest;
+
+        /// <summary>
+        /// Sets the OAuth details of the request.
+        /// </summary>
+        /// <param name="redirectUrl">
+        /// The redirect url for the request's oauth details, must not be blank.
+        /// </param>
+        /// <param name="scope">
+        /// The scope for the request's oauth details, must not be blank.
+        /// </param>
+        /// <param name="state">
+        /// The state for the request's oauth details.
+        /// </param>
+        /// <returns>
+        /// A reference to the <see cref="AddToCalendarRequestBuilder"/>.
+        /// </returns>
+        /// <exception cref="ArgumentException">
+        /// Thrown if <paramref name="redirectUrl"/> or <paramref name="scope"/> are empty.  
+        /// </exception>
+        public AddToCalendarRequestBuilder OAuthDetails(string redirectUrl, string scope, string state = null)
+        {
+            Preconditions.NotBlank("redirectUrl", redirectUrl);
+            Preconditions.NotBlank("scope", scope);
+
+            this.oauth = new AddToCalendarRequest.OAuthDetails
+            {
+                RedirectUrl = redirectUrl,
+                Scope = scope,
+                State = state
+            };
+
+            return this;
+        }
+
+        /// <summary>
+        /// Sets the event details of the request.
+        /// </summary>
+        /// <param name="upsertEventRequestBuilder">
+        /// The event details builder for the request, must not be null.
+        /// </param>
+        /// <returns>
+        /// A reference to the <see cref="AddToCalendarRequestBuilder"/> 
+        /// </returns>
+        public AddToCalendarRequestBuilder UpsertEventRequestBuilder(UpsertEventRequestBuilder upsertEventRequestBuilder)
+        {
+            Preconditions.NotNull("event", upsertEventRequestBuilder);
+
+            this.upsertEventRequestBuilder = upsertEventRequestBuilder;
+
+            return this;
+        }
+
+        /// <summary>
+        /// Sets the event details of the request.
+        /// </summary>
+        /// <param name="upsertEventRequest">
+        /// The event details for the request, must not be null.
+        /// </param>
+        /// <returns>
+        /// A reference to the <see cref="AddToCalendarRequestBuilder"/>.
+        /// </returns>
+        public AddToCalendarRequestBuilder UpsertEventRequest(UpsertEventRequest upsertEventRequest)
+        {
+            Preconditions.NotNull("event", upsertEventRequest);
+
+            this.upsertEventRequest = upsertEventRequest;
+
+            return this;
+        }
+
+        /// <inheritdoc />
+        public AddToCalendarRequest Build()
+        {
+            var upsertEventRequest = this.upsertEventRequestBuilder != null 
```

Comment:
As a general rule, don't put a "not" in a ternary, swap the other two arguments instead.

```cs
// Instead of
!a ? x : y

// Do
a ? y : x
```

It's the same kind of thing as not using `unless-else` in Ruby.